docs/ARCHITECTURE.md

```diff
@@ -0,0 +1,50 @@
+# ARCHITECTURE
+
+## Primitive handling
+
+ActiveModelSerializers doesn't handle primitives, such as a String or Array, passed to `render json:`.
+When an resource being serialized has an assocation that is a primitive, or is a primitive
+in a collection, it is passed through without modification. (To be precise, this applies to any resource
+without a defined serializer.)
```

Comment:
The wording is slightly confusing, although I cannot come up with a better solution right now.